### DISCUSSION TITLE
Share my energy based model tutorial

### DISCUSSION THREAD

**Discussion by odie2630463** [google/flax#1348] [Show and tell]
Hi Flax, I implement [EBM in Flax](https://colab.research.google.com/drive/1g0bWCn-EkEa_DWdnQkvkfWaNOe5nywq4?usp=sharing&fbclid=IwAR1Fn0N0UDSj1e9VOCOmfm_4_vCUh79cJrEkNf14N6ST1lJAKHGQ6DOkb-c), share with everyone, and if has any advice I would like to know. Thanks!

**Comment by avital** [comment]
Cool! I don't know much about EBMs but took quick glace at your Colab -- why not make swish simply a function? No need to make it a module as it has no variables or uses no RNGs.

**Reply by odie2630463** [reply]
Hi, EBM is the energy-based model for fitting data distribution P(X). Swish roll is an unknown distribution and we only have the data samples from it. So we training the model on the data and synths, which is sampled by MCMC sampler from the model. Flax version is faster than Pytorch 50x. Awesome !!!
